### PR TITLE
Switch to Plone Sphinx Theme for documentation, include changes to http-examples to build docs preview

### LINKED ISSUE #1097: Browser tiltle of documentation still says version 1.0a1
https://plonerestapi.readthedocs.io/en/latest/searching.html

https://github.com/plone/plone.restapi/blob/03148b67af8026827b69a812796b50ae273c5857/docs/source/conf.py#L74

### PR DESCRIPTION
**Pull request by stevepiercy**
- Switch to Plone Sphinx Theme for documentation
- The `http-examples` directory is now included for pull request previews.
- I resynced the RTD webhook in this repo's settings as the PR preview builds were failing.

Closes #1097

----
📚 Documentation preview 📚: https://plonerestapi--1815.org.readthedocs.build/

### CODE REVIEW

**Comment by stevepiercy** [comment]
@jenkins-plone-org please run jobs

**Comment by stevepiercy** [comment]
There are two issues with the theme, which I can resolve in a future pull request, as this is an improvement.

1. The tabs for http-examples have a white background in both light and dark mode, but should adapt.
2. ScrollSpy is not working on the right-hand in-page navigation.

File: docs/source/conf.py [hunk 1]
```diff
@@ -159,24 +161,69 @@ def patch_pygments_to_highlight_jsonschema():
...
 # The theme to use for HTML and HTML Help pages.  See the documentation for
 # a list of builtin themes.
-html_theme = "sphinx_book_theme"
-
+html_theme = "plone_sphinx_theme"
 html_logo = "_static/logo.svg"
 html_favicon = "_static/favicon.ico"
 
-html_css_files = ["custom.css", ("print.css", {"media": "print"})]
+# html_css_files = ["custom.css", ("print.css", {"media": "print"})]
```
Comment (wesleybl) [inline docs/source/conf.py:168]: If you are not going to use it, it is better to remove this line.

**Comment by wesleybl** [comment]
> The tabs for http-examples have a white background in both light and dark mode, but should adapt.

This really looks weird, no contrast. It really needs to be fixed.

File: docs/source/conf.py (hunk 1 above)
Comment (stevepiercy) [reply docs/source/conf.py]: Done in https://github.com/plone/plone.restapi/pull/1815/commits/4be1632dc2b86e013153b2dbfcb69b2abfc4e962

**Comment by stevepiercy** [comment]
> > The tabs for http-examples have a white background in both light and dark mode, but should adapt.
> 
> This really looks weird, no contrast. It really needs to be fixed.

This should be fixed in [PST 0.0.15](https://github.com/plone/plone-sphinx-theme/releases/tag/0.0.15). Preview: https://plonerestapi--1815.org.readthedocs.build/en/1815/endpoints/aliases.html#adding-new-url-aliases-for-a-page

**Comment by stevepiercy** [comment]
The ScrollSpy bug is weird. Something in this repo interferes with it, as it is fine in the [PST docs](https://plone-sphinx-theme.readthedocs.io/guides/contribute.html#edit-documentation). If anyone can spy the issue, I'd appreciate the assist.

**Comment by stevepiercy** [comment]
@jenkins-plone-org please run jobs

**Comment by stevepiercy** [comment]
I'm not sure why the one Jenkins run fails, as this PR has nothing to do with its failing tests.

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -60,7 +60,7 @@ jobs:
 
       # build sphinx
       - name: sphinx
-        run: if [ "${{ matrix.plone-version }}" == "6.0" ] && [ ${{ matrix.python-version }} == '3.12' ]; then make docs-html; fi
+        run: if [ "${{ matrix.plone-version }}" == "6.0" ] && [ ${{ matrix.python-version }} == '3.12' ]; then pip install -r requirements-docs.txt && make docs-html; fi
```
Comment (davisagli) [inline .github/workflows/tests.yml:63]: @stevepiercy If I'm reading the Makefile correctly, `make docs-html` already depends on `bin/python`, which runs this pip install. So it shouldn't be necessary to do it explicitly here.

**Review by davisagli — APPROVED** [review]

File: .github/workflows/tests.yml (hunk 2 above)
Comment (stevepiercy) [reply .github/workflows/tests.yml:63]: Good catch. Thank you! Supplemental PR https://github.com/plone/plone.restapi/pull/1817

**Pull request merged**